File: dev.pitlor.SummerCamp/Models/Game.cs
using System.Collections.Immutable;

namespace dev.pitlor.SummerCamp.Models;

public record struct Game(
    Deck Deck1,
    Deck Deck2,
    Deck Deck3,
    int SmoresLeft,
    int ScavengerHuntLeft,
    int FreeTimeLeft,
    ImmutableList<string> PlayerOrder,
    Dictionary<string, Player> Players,
    ImmutableList<BoardTile> BoardTiles,
    ImmutableList<Badge> ParticipationBadges,
    ImmutableList<Badge> AllStarBadges,
    string AdminPlayerId,
    bool IsStarted,
    string CurrentPlayerId);

File: dev.pitlor.SummerCamp.Client/src/models/game.ts
import type { Deck } from "./deck.ts";
import type { BoardTile } from "./board-tile.ts";
import type { Player } from "./player.ts";

export interface Game {
  deck1: Deck;
  deck2: Deck;
  deck3: Deck;
  smoresLeft: number;
  scavengerHuntLeft: number;
  freeTimeLeft: number;
  playerOrder: string[];
  players: Record<string, Player>;
  boardTiles: BoardTile[];
  adminPlayerId: string;
  isStarted: boolean;
}
